proposals/reward-proposal-v2-reward-distribution.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Epoch Reward Distribution
+---
+
+## Problem
+
+In the `Epoch Reward Calculation V2`, we proposed a new method to handle
+calculating epoch rewards. This is the second part of that proposal - `Reward
+Distribution`.
+
+## Proposed Solutions
+
+Similar to [Partitioned Inflationary Rewards
+Distribution](https://github.com/solana-labs/solana/blob/master/docs/src/proposals/partitioned-inflationary-rewards-distribution.md),
+the rewards are distributed over `M` slots.
+
+To address the concern for the lack of double-entry booking for reward
+distribution, a special system instruction, `DistributeReward`, will be
+introduced to distribute the rewards from the family of `EpochRewardReserve`
+accounts and the `EpochRewardProof` system account.
+
+This instruction takes the target reward account, the amount to distribute, the
+`EpochRewardReserve` account, and the merkle proof for the distribution. When
+executing this instruction, it first verifies the supplied merkle proof against
+the proof stored in in `EpochRewardProof`. If the verification passed, it will
+transfer the reward from the `EpochRewardReserve` account to the target reward
+account.
+
+To spread out the reward distribution, for each slot, a fixed number (`K`) of
+stake/vote accounts's rewards are distributed. The leaders, within `M`
+slot-height, will be responsible for distributing the rewards. Ideally, these
```

Comment:
* What happens if `M` is not enough with fixed `K` per slot?
* Is `K` in terms number of vote or stake accounts? Stake vs vote accounts have a very asymmetric load.
* What happens if there are a lot of skipped slots and the slots are not rooted?
  * Does this make any assumption that at least certain number of slots are rooted during each epoch?
  * What happens if the cluster goes down during these rewards distribution phase?
     * In particular are there any parts of this computation only stored on memory which are lost if the node goes down? Like I am still not sure what happens to merkle tree leaves.